addons/l10n_account_edi_ubl_cii_tests/tests/common.py

```diff
@@ -0,0 +1,179 @@
+
+    @classmethod
+    def _get_tax_by_xml_id(cls, tax_xml_id):
+        """ Helper to retrieve a tax easily.
+
+        :param tax_xml_id:  The tax template's xml id.
+        :return:            An account.tax record
+        """
+        module, trailing_id = tax_xml_id.split('.')
+        return cls.env.ref(f'{module}.{cls.env.company.id}_{trailing_id}')
+
+    def assert_same_invoice(self, invoice1, invoice2, **invoice_kwargs):
+        self.assertEqual(len(invoice1.invoice_line_ids), len(invoice2.invoice_line_ids))
+        self.assertRecordValues(invoice2, [{
+            'partner_id': invoice1.company_id.partner_id.id,
+            'invoice_date': fields.Date.from_string(invoice1.date),
+            'currency_id': invoice1.currency_id.id,
+            'amount_untaxed': invoice1.amount_untaxed,
+            'amount_tax': invoice1.amount_tax,
+            'amount_total': invoice1.amount_total,
+            **invoice_kwargs,
+        }])
+
+        default_invoice_line_kwargs_list = [{}] * len(invoice1.invoice_line_ids)
+        invoice_line_kwargs_list = invoice_kwargs.get('invoice_line_ids', default_invoice_line_kwargs_list)
+        self.assertRecordValues(invoice2.invoice_line_ids, [{
+            'quantity': line.quantity,
+            'price_unit': line.price_unit,
+            'discount': line.discount,
+            'product_id': line.product_id.id,
+            'tax_ids': line.tax_ids.ids,
+            'product_uom_id': line.product_uom_id.id,
+            **invoice_line_kwargs,
+        } for line, invoice_line_kwargs in zip(invoice1.invoice_line_ids, invoice_line_kwargs_list)])
+
+    # -------------------------------------------------------------------------
+    # IMPORT
+    # -------------------------------------------------------------------------
+
+    @freeze_time('2017-01-01')
+    def _import_invoice(self, invoice, xml_etree, xml_filename):
+        new_invoice = self.edi_format._create_invoice_from_xml_tree(
+            xml_filename,
+            xml_etree,
+            self.company_data['default_journal_purchase'],
+        )
+
+        self.assertTrue(new_invoice)
+        self.assert_same_invoice(invoice, new_invoice)
+
+    def _import_invoice_from_file(self, subfolder, filename, amount_total, amount_tax, move_type='in_invoice', currency='EUR'):
+        # import the file as a new invoice / update the invoice from file
+        # assert that the amount_total, amount_tax, currency is correct
+        invoice = self._create_empty_vendor_bill()
+        invoice.move_type = move_type
+        invoice_count = len(self.env['account.move'].search([]))
+        self.update_invoice_from_file('l10n_account_edi_ubl_cii_tests', subfolder, filename, invoice)
+
+        self.assertEqual(len(self.env['account.move'].search([])), invoice_count)
+
+        self.assertEqual(invoice.amount_total, amount_total)
+        self.assertEqual(invoice.amount_tax, amount_tax)
+        self.assertEqual(invoice.currency_id.name, currency)
```

Comment:
use `assertRecordValues`
Also, `self.assertEqual(invoice.currency_id.name, currency)` is weird... currency is not a currency but a string?